index.js

```diff
@@ -98,7 +98,12 @@ UglifyWriter.prototype.processFile = function(inFile, outFile, relativePath, out
 
     if (srcURL.existsIn(src)) {
       let url = srcURL.getFrom(src);
-      sourceMap.content = JSON.parse(fs.readFileSync(path.join(path.dirname(inFile), url)));
+      let sourceMapPath = path.join(path.dirname(inFile), url);
+      if (fs.existsSync(sourceMapPath)) {
+        sourceMap.content = JSON.parse(fs.readFileSync(sourceMapPath));
+      } else {
+        console.warn(`[WARN] (broccoli-uglify-sourcemap) "${url}" referenced in "${relativePath}" could not be found`);
```

Comment:
Can you modify this to use `this.console.warn` (then initialize `this.console = options.console || console` in the constructor)? There is at least one other usage of `console.warn` also.

The idea is to have ember-cli-uglify pass in a “custom console” which uses our `UI` object and avoiding needing to manually check for `--silent`...

Reply:
yeah, I noticed that too, but wanted to address it in a separate PR. your point about honoring `--silent` is valid though, this should probably be something like `} else if (!this.silent) {`.

Reply:
@rwjblue adjusted